generator/generator_tests/feature_builder_test.cpp

```diff
@@ -218,6 +220,54 @@ UNIT_CLASS_TEST(TestWithClassificator, FBuilder_ParamsParsing)
   TEST_EQUAL(params.house.Get(), "0", ());
 }
 
+UNIT_CLASS_TEST(TestWithClassificator, FBuilder_Housenumbers)
+{
+  FeatureBuilderParams params;
+  std::string_view name;
+
+  params.MakeZero();
+  name = {};
+  params.AddHouseName("75a"); // addr:housenumber
+  params.AddHouseName("Sandpiper"); // addr:housename
+  TEST_EQUAL(params.house.Get(), "75a", ());
+  TEST(params.name.GetString(StringUtf8Multilang::kDefaultCode, name), ());
+  TEST_EQUAL(name, "Sandpiper", ());
+
+  params.MakeZero();
+  name = {};
+  params.AddHouseName("75"); // addr:housenumber
+  params.AddHouseName("Bld 2"); // addr:housename
+  TEST_EQUAL(params.house.Get(), "75", ());
+  TEST(params.name.GetString(StringUtf8Multilang::kDefaultCode, name), ());
+  TEST_EQUAL(name, "Bld 2", ());
+
+  params.MakeZero();
+  name = {};
+  params.AddHouseName("75"); // addr:housenumber
+  params.AddHouseName("2"); // addr:housename
+  TEST_EQUAL(params.house.Get(), "2", ());
+  TEST(params.name.GetString(StringUtf8Multilang::kDefaultCode, name), ());
+  TEST_EQUAL(name, "75", ());
+
+  params.MakeZero();
+  name = {};
+  params.name.AddString(StringUtf8Multilang::kDefaultCode, "The Mansion");
+  params.AddHouseName("75a"); // addr:housenumber
+  params.AddHouseName("Sandpiper"); // addr:housename
+  TEST_EQUAL(params.house.Get(), "75a", ());
+  TEST(params.name.GetString(StringUtf8Multilang::kDefaultCode, name), ());
+  TEST_EQUAL(name, "The Mansion", ());
+
+  params.MakeZero();
+  name = {};
+  params.AddHouseName("75a"); // addr:housenumber
+  params.AddHouseName("2"); // addr:housename
+  params.name.AddString(StringUtf8Multilang::kDefaultCode, "The Mansion");
+  TEST_EQUAL(params.house.Get(), "2", ());
+  TEST(params.name.GetString(StringUtf8Multilang::kDefaultCode, name), ());
+  TEST_EQUAL(name, "The Mansion", ());
+}
```

Comment:
Please, remove your changes here, oms2type, feature_data.
This logic is reworked in generator branch and I will get unnecessary merge conflicts.
It will be in a PR soon.

Anyway, what is the reason to check 2 sequential AddHouseName calls? It won't happen in the generator anyway.